exercises/chapter8/src/Data/AddressBook.purs

```diff
@@ -17,11 +19,10 @@ data PhoneType
   | CellPhone
   | OtherPhone
 
+derive instance genericPhoneType :: Generic PhoneType _
+
 instance showPhoneType :: Show PhoneType where
-  show HomePhone  = "HomePhone"
-  show WorkPhone  = "WorkPhone"
-  show CellPhone  = "CellPhone"
-  show OtherPhone = "OtherPhone"
+  show = genericShow
 
```

Comment:
Same total number of lines, but the generic version is a lot more maintainable. Hopefully this doesn't add too much noise / confusion / complexity for readers.